### ISSUE TITLE
Hand Press & Die Set (UPS) doesn't work

### ISSUE THREAD

**Issue by jim-huynh** [CleverRaven/Cataclysm-DDA#18907]
Version: 0.C-19262-g6cb1f91

The hand press & die set doesn't work when there is a UPS mod attached.  The hand press & die set using batteries works just fine.

![hand press die set](https://cloud.githubusercontent.com/assets/3409545/19545011/e7bf4a62-9637-11e6-9970-f30dd1d22aae.png)

**Comment by kevingranade** [comment]
Looks like the same issue as #18912

**Comment by l-Ashery-l** [comment]
Not so sure about that. That issue looks to be related to recognizing that there's an available UPS to transfer/contribute charge, while this is about the tool flat out not being recognized by the game once modded. Not only that, but the UPS modified tool no longer gets recognized as a component, which is how I first ran into this issue when attempting to make a FoodCo buddy using a modded press and die set.

Another way to see this is with a gunsmith repair kit. Attach a UPS mod to it and let it charge up a bit and you'll be able to craft everything that requires charge, but you _won't_ be able to do the Mosin-Nagant rifle conversions since those require you to simply have the tool in your possession.

The same bug was posted at #19457, though you appear to have checked that just a couple days ago so that's nothing new to you.

Not knowing how the game's inner mechanics work, my guess is that there's a straight name check and the extra "+" at the end throws things off. However, that theory breaks down as I'm able to use some modded light gloves (Fur lined) to craft survival light gloves. But maybe the different types of mods don't quite work under the hood in the same way.

The component issue is definitely _not_ restricted to UPS modded items. Adding the extra battery mod to a food dehydrator made it ineligible to be used as part of a FoodCo.

The same thing appears when you try to convert a modded Mosin-Nagant rifle. In that case, however, you likely would _want_ the game to prevent you from converting a modded rifle to an unmodded one, destroying your mods in the process. Maybe this issue is entirely a side effect of a deliberately designed fail safe to prevent the player from destroying mods on a weapon by accident.

PBA masks also can't be used to craft a survival mask when modded for extra batteries, but one can be used when padded with leather.

Apologies if I went a bit off course from this thread. I'm just poking around in game and trying to test what I can to see if there's some insight that can help the devs pinpoint where the issue lies.
